src/phasorpy/io.py

```diff
@@ -522,6 +540,266 @@ def read_sdt(
+            raise ValueError(
+                f'{os.path.basename(filename)!r} '
+                'is not a PTU file containing a T3 mode image'
+            )
+        data = ptu.decode_image(
+            selection,
+            dtype=dtype,
+            frame=frame,
+            channel=channel,
+            keepdims=keepdims,
+            asxarray=True,
+        )
+
+    assert isinstance(data, DataArray)
+    data.attrs['frequency'] *= 1e-6  # MHz
+    return data
+
+
+def read_flif(
+    filename: str | PathLike[Any],
+    /,
+) -> DataArray:
+    """Return frequency-domain image and metadata from FlimFast FLIF file.
+
+    FlimFast FLIF files contain camera images and metadata from
+    frequency-domain fluorescence lifetime measurements.
+
+    Parameters
+    ----------
+    filename : str or Path
+        Name of FlimFast FLIF file to read.
+
+    Returns
+    -------
+    xarray.DataArray
+        Frequency-domain phase images with :ref:`axes codes <axes>` ``'THYX'``
+        and type ``uint16``:
+
+        - ``coords['H']``: phases in radians.
+        - ``attrs['frequency']``: repetition frequency in MHz.
+        - ``attrs['ref_phase']``: measured phase of reference.
+        - ``attrs['ref_mod']``: measured modulation of reference.
+        - ``attrs['ref_tauphase']``: lifetime from phase of reference.
+        - ``attrs['ref_taumod']``: lifetime from modulation of reference.
+
+    Raises
+    ------
+    lfdfile.LfdFileError
+        File is not a FlimFast FLIF file.
+
+    Examples
+    --------
+    >>> data = read_flif(fetch('flimfast.flif'))
+    >>> data.values
+    array(...)
+    >>> data.dtype
+    dtype('uint16')
+    >>> data.shape
+    (32, 220, 300)
+    >>> data.dims
+    ('H', 'Y', 'X')
+    >>> data.coords['H'].data
+    array(...)
+    >>> data.attrs['frequency']
+    80.65...
+
+    """
+    import lfdfiles
+
+    with lfdfiles.FlimfastFlif(filename) as flif:
+        nphases = int(flif.header.phases)
+        data = flif.asarray()
+        if data.shape[0] < nphases:
+            raise ValueError(f'measured phases {data.shape[0]} < {nphases=}')
+        if data.shape[0] % nphases != 0:
+            data = data[: (data.shape[0] // nphases) * nphases]
+        data = data.reshape(-1, nphases, data.shape[1], data.shape[2])
+        if data.shape[0] == 1:
+            data = data[0]
+            axes = 'HYX'
+        else:
+            axes = 'THYX'
+        # TODO: check if phases are ordered
+        phases = numpy.radians(flif.records['phase'][:nphases])
+        metadata = _metadata(axes, data.shape, H=phases)
+        attrs = metadata['attrs']
+        attrs['frequency'] = float(flif.header.frequency)
+        attrs['ref_phase'] = float(flif.header.measured_phase)
+        attrs['ref_mod'] = float(flif.header.measured_mod)
+        attrs['ref_tauphase'] = float(flif.header.ref_tauphase)
+        attrs['ref_taumod'] = float(flif.header.ref_taumod)
+
+    return DataArray(data, **metadata)
+
+
+def read_fbd(
```

Comment:
This function also worked perfectly. I tested with different data and returned the expected data. I think it would be great if it can also be implemented the 'frame', 'channel' and 'keepdims' arguments like in the read_ptu function so that the returned image already has only three dimensions.

Reply:
I added/changed the `frame`, `channel`, and `keepdims` parameters to the `read_fbd` function to make it more alike to `read_ptu`.

Reply:
Thanks for adding this options, I tested them in read_fbd and they work great and helps to speed up the process of importing the data intro the correct format.